mon-pix/tests/unit/services/focused-certification-challenges-manager_test.js

```diff
@@ -0,0 +1,128 @@
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+import sinon from 'sinon';
+import { setupTest } from 'ember-mocha';
+import { A } from '@ember/array';
+
+describe('Unit | Service | focused-certification-challenges-manager', function () {
+  setupTest();
+
+  const originalGetItem = window.localStorage.getItem;
+  const originalSetItem = window.localStorage.setItem;
+  const originalRemoveItem = window.localStorage.removeItem;
+
+  beforeEach(function () {
+    window.localStorage.getItem = sinon.stub();
+    window.localStorage.setItem = sinon.stub();
+    window.localStorage.removeItem = sinon.stub();
+  });
+
+  after(function () {
+    window.localStorage.getItem = originalGetItem;
+    window.localStorage.setItem = originalSetItem;
+    window.localStorage.removeItem = originalRemoveItem;
+  });
+
+  describe('#constructor', function () {
+    it('should instantiate an empty Ember Array if there is no focus challenges in the local storage', function () {
+      // given
+      window.localStorage.getItem.withArgs('focusedCertificationsChallenges').returns(null);
+
+      // when
+      const service = this.owner.lookup('service:focused-certification-challenges-manager');
+
+      // then
+      expect(service._focusedCertificationsChallenges).to.deep.equal(A([]));
+    });
+
+    it('should retrieve focus challenges in the local storage if there is any', function () {
+      // given
+      window.localStorage.getItem
+        .withArgs('focusedCertificationsChallenges')
+        .returns('["challengeId1", "challengeId2"]');
+
+      // when
+      const service = this.owner.lookup('service:focused-certification-challenges-manager');
+
+      // then
+      expect(service._focusedCertificationsChallenges).to.deep.equal(A(['challengeId1', 'challengeId2']));
+    });
+  });
+
+  describe('#add', function () {
+    it('should add a focused challengeId to local storage', function () {
+      // given
+      window.localStorage.getItem.withArgs('focusedCertificationsChallenges').returns('["challengeId1"]');
+      const service = this.owner.lookup('service:focused-certification-challenges-manager');
+
+      // when
+      service.add('challengeId2');
+
+      // then
+      sinon.assert.calledWith(
```

Comment:
**suggestion non gloquante :** Je testerai le bon fonctionnement du add via le has, pour tester les interfaces et pas l'implémentation.
`expect(service.has('challengeId2').to.be.true;`
(pardon si pas la bonne syntaxe, je me remémore petit à petit :D )